runtime/platform/mach_o.h

```diff
@@ -44,6 +68,73 @@ struct mach_header_64 {
 static constexpr uint32_t MH_MAGIC_64 = 0xfeedfacf;
 static constexpr uint32_t MH_CIGAM_64 = 0xcffaedfe;
 
+static constexpr uint32_t LC_SYMTAB = 0x2;
+static constexpr uint32_t LC_SEGMENT_64 = 0x19;
+
+struct symtab_command {
+  uint32_t cmd;
+  uint32_t cmdsize;
+  uint32_t symoff;  // Offset to the symbol table, relative to beginning of the
+                    // loaded linkedit segment.
+  uint32_t nsyms;
+  uint32_t stroff;   // Offset to the string table, relative to beginning of the
+                     // loaded linkedit segment.  The string table immediately
+                     // follows the symbol table and is just an series of
+                     // null-terminated strings one after the other.
+  uint32_t strsize;  // Size of the string table.
+};
+
+// Symbol table entry.
+struct nlist_64 {
+  union {
+    uint32_t n_strx;  // index into the string table.
+  } n_un;
+  uint8_t n_type;
+  uint8_t n_sect;  // Section number (1-based).
+  uint16_t n_desc;
+  uint64_t n_value;  // Value of this symbol (e.g. address)
+};
+
+// A Segment Load Command.  In a Dart snapshot, there will only be 3 segments:
+// __TEXT, __DATA, and __LINKEDIT.
+// The __TEXT segment contains the code and read-only data.
+// The __DATA segment contains the read-write data.
+// The __LINKEDIT segment contains the symbol table and string table.
+struct segment_command_64 {
+  uint32_t cmd;
+  uint32_t cmdsize;
+  char segname[16];
+  uint64_t vmaddr;
+  uint64_t vmsize;
+  uint64_t fileoff;
+  uint64_t filesize;
+  vm_prot_t maxprot;
+  vm_prot_t initprot;
+  uint32_t nsects;
+  uint32_t flags;
+};
+
+static constexpr uint32_t VM_PROT_NONE = 0x00;
+static constexpr uint32_t VM_PROT_READ = 0x01;
+static constexpr uint32_t VM_PROT_WRITE = 0x02;
+static constexpr uint32_t VM_PROT_EXECUTE = 0x04;
+
+// Segments may contain zero or more sections.
+struct section_64 {
```

Comment:
I didn't actually end up needing segment parsing, so this probably can be removed too.